examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform abstraction for radio communication.
+ *
+ */
+
+#include <openthread/config.h>
+#include <openthread/platform/alarm-milli.h>
+#include <openthread/platform/diag.h>
+#include <openthread/platform/radio.h>
+
+#include "common/logging.hpp"
+#include "utils/code_utils.h"
+#include "utils/mac_frame.h"
+#include "utils/soft_source_match_table.h"
+
+#include "platform-eagle.h"
+
+enum
+{
+    IEEE802154_MIN_LENGTH      = 5,
+    IEEE802154_MAX_LENGTH      = 127,
+    IEEE802154_ACK_LENGTH      = 5,
+    IEEE802154_FRAME_TYPE_MASK = 0x7,
+    IEEE802154_FRAME_TYPE_ACK  = 0x2,
+    IEEE802154_FRAME_PENDING   = 1 << 4,
+    IEEE802154_ACK_REQUEST     = 1 << 5,
+    IEEE802154_DSN_OFFSET      = 2,
+};
+
+typedef enum
+{
+    PHY_CCA_IDLE    = 0x04,
+    PHY_CCA_TRX_OFF = 0x03,
+    PHY_CCA_BUSY    = 0x00,
+} phy_ccaSts_t;
+
+unsigned char tx_buffer[256] __attribute__((aligned(4)));
+
+unsigned int r_ptr = 0;
+unsigned int w_ptr = 0;
+
+unsigned char current_channel;
+#define LOGICCHANNEL_TO_PHYSICAL(p) (((p)-10) * 5)
+
+static otExtAddress   sExtAddress;
+static otShortAddress sShortAddress;
+static otPanId        sPanid;
+static int8_t         sTxPower = 0;
+
+static otRadioFrame sTransmitFrame;
+
+otRadioFrame        sAckFrame;
+static otError      sTransmitError;
+static otError      sReceiveError;
+static uint8_t      sTransmitPsdu[IEEE802154_MAX_LENGTH];
+static uint8_t      sAckPsdu[8];
+static otRadioState sState           = OT_RADIO_STATE_DISABLED;
+static bool         sSrcMatchEnabled = false;
+volatile int        tx_busy          = 0;
+
+#define RX_FRAME_SLOT_NUM 6
+#define MEM_ALIGNMENT 4
+#define MEM_ALIGN_SIZE(size) (((size) + MEM_ALIGNMENT - 1) & ~(MEM_ALIGNMENT - 1))
+unsigned char rx_buffer[160] __attribute__((aligned(4)));
+unsigned char rx_frame_slots[RX_FRAME_SLOT_NUM][MEM_ALIGN_SIZE(sizeof(otRadioFrame)) + 160] __attribute__((aligned(4)));
+typedef int   semaphore;
+semaphore     empty = RX_FRAME_SLOT_NUM;
+semaphore     full  = 0;
+unsigned char rx_frame[MEM_ALIGN_SIZE(sizeof(otRadioFrame)) + 160] __attribute__((aligned(4)));
+
+/**
+ * Get the radio capabilities.
+ *
+ * @param[in] aInstance  The OpenThread instance structure.
+ *
+ * @returns The radio capability bit vector (see `OT_RADIO_CAP_*` definitions).
+ *
+ */
```

Comment:
Could we remove the comment from this file? The header file has already included these comments.